### PR TITLE
More vampire additions

### PR DESCRIPTION
**Pull request by SariusSkelrets**
#### Summary
Mods "(XE) add more vampire traits, allow to drink from friends and make blood of saints actually block vampirism"

#### Purpose of change

Tier 4 needs more traits. I'm also adding more vampire-related stuff such as drinking from allies as killing and bleeding ferals all the time is impractical

#### Describe the solution

Add the following:

- [x] The ability to drink from allied NPCs. Kinda-friends won't like it, true friends won't mind it and others won't allow it. It takes some days before they allow you to do it again
- [x] Make blood of saints actually block vampirism

And these powers:
- [x] Sanguine Ecstasy (Tier 1: make drinking human blood make you feel great)
- [x] Blood-Fueled Magic (Tier 3: blood-to-mana conversion ability, scales with your vampiric power)
- [x] Torpor (Tier 4: can enter torpor, in which you heal faster, gain the benefits of sleep and pasively consume less blood. The drawback is that you are asleep during that time and nothing can wake you up)
- [x] True Vampire's Potence (Tier 4: counts as six tier 4 traits for the purpose of amplifying other traits)
- [x] Dominating Gaze (Tier 4: temporary mind control for targets with a mind, efficiency and duration scales with your vampiric power, doesn't work on NPCs)
- [x] Corpse-like Flesh (Tier 4: make you immune to disease, infection, parasites and radiation)
- [x] Facultative Breathing (Tier 4: makes you unable to drown and unaffected by inhaled gases as your lungs are mostly vestigial now)

#### Describe alternatives you've considered

Add other vampire traits

I've considered adding a way to drink from stunned humans or after succeeding at an intimidation check, which would both massively piss them off, but until I find how to make it work with ferals and non-npc civillans, it'll have to wait

#### Testing
drinking from a NPC grants you blood and gives them the effect that checks how long it has been since you last did
NPCs that like you enough won't mind
drinking from a NPC that doesn't like you enough lower their opinion of you
A low enough opinion of you make them refuse
NPCs with the effect will refuse to let you drink more until it goes away
The mutations and flags that prevent star vampires from drinking someone's blood also keep you from trying to do the same
only one drinking dialogue appears at once

Blood of saints makes you immune to vampvirus and will eventualy cure you if you're not too much of a vampire

Sanguine ecstasy grants a mood buff when drinking blood, whether by item or dialogue
Blood-fueled magic grant a spell that costs blood and gives mana
Torpor makes you sleep for eight hours, healing you faster, fully recovering your sleep and reducing your blood consumption. You cannot wake up before that time is up
True vampire's potence boosts other vampiric abilities
Dominating gaze made a feral become an ally
Corpse-like flesh protects against infections, diseases, parasites and radiation
Facultative breathing protects against drowing and inhaled gases and nothing else 

#### Additional context

Might decide to add more traits to this PR, if I can find more stuff that fits

Until NPCs can be made to be temporarily friendly, dominating gaze won't work on them

### CODE REVIEW

File: data/mods/Xedra_Evolved/mutations/mutations.json [hunk 1]
```diff
@@ -445,6 +466,112 @@
...
+    "type": "mutation",
+    "id": "TRUE_VAMPIRE_POWER_5",
+    "points": 0,
+    "player_display": false,
+    "starting_trait": false,
+    "purifiable": false,
+    "valid": false,
+    "name": { "str": "True Vampire's Potence 5" },
+    "description": "Fake mutation to boost vampiric abilities."
+  },
```
Comment (Standing-Storm) [inline data/mods/Xedra_Evolved/mutations/mutations.json:540]: If you're planning on inserting these into the Jmath, you don't need 5 separate mutations to do it. You can just do 

> {
>     "type": "jmath_function",
>     "id": "vampire_total_tier_four_traits",
>     "num_args": 0,
>     "return": "(u_has_trait('TRUE_VAMPIRE_POWER') * 5)"
>   }

...and True Vampire's Potence will count as five mutations' worth for the scaling math.

Comment (SariusSkelrets) [reply data/mods/Xedra_Evolved/mutations/mutations.json:540]: Good point. I'll begin setting that up in a soon-ish push

File: data/mods/Xedra_Evolved/items/armor/integrated.json [hunk 2]
```diff
@@ -793,5 +793,21 @@
       }
     ],
     "armor": [ { "covers": [ "torso" ], "coverage": 0, "encumbrance": 0 } ]
+  },
+  {
+    "id": "integrated_nobreathe",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": { "str_sp": "vestigial lungs" },
+    "description": "Breathing is now something you decide to do, rather than a biological necessity.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/armor/integrated.json:802]: Does this autoset oxygen levels so that you can't asphyxiate from being crushed to death?

Comment (SariusSkelrets) [reply data/mods/Xedra_Evolved/items/armor/integrated.json:802]: I first thought that the oxygen value were hardcoded

Then I checked and the REBREATHER flag exists and prevents being suffocated by crushing so I'll add it to the lungs

The messages are hardcoded however, so they'll get told that they are suffocating regardless

Comment (Maleclypse) [reply data/mods/Xedra_Evolved/items/armor/integrated.json:802]: Ok that’ll create an issue that will hopefully result in oxygen getting less hardcoded.

**Pull request merged**